pkg/segment/aggregations/segaggs.go

```diff
@@ -1937,14 +1939,55 @@ func performMultiValueColRequestWithoutGroupby(letColReq *structs.LetColumnsRequ
 		case "makemv":
 			finalValue := performMakeMV(fieldValueStr, mvColReq)
 			rec[mvColReq.ColName] = finalValue
+		case "mvexpand":
+			expandedRecs := performMVExpand(fieldValue)
+
+			// Apply limit if mvColReq.Limit is greater than 0
+			if mvColReq.Limit > 0 && len(expandedRecs) > int(mvColReq.Limit) {
+				expandedRecs = expandedRecs[:mvColReq.Limit]
+			}
+			delete(recs, key)
+			for i, expandedValue := range expandedRecs {
+				expandedValueStr, ok := expandedValue.(string)
+				if !ok {
+					expandedValueStr = fmt.Sprintf("%v", expandedValue)
+				}
+				newRec := make(map[string]interface{})
+				for k, v := range rec {
+					newRec[k] = v
+				}
+				newRec[mvColReq.ColName] = expandedValueStr
+				newRecs[fmt.Sprintf("%s_%d", key, i)] = newRec
+			}
 		default:
 			return fmt.Errorf("performMultiValueColRequestWithoutGroupby: unknown command %s", mvColReq.Command)
 		}
 	}
-
+	if mvColReq.Command == "mvexpand" {
+		for k, v := range newRecs {
+			recs[k] = v
+		}
+	}
 	return nil
 }
 
+func performMVExpand(fieldValue interface{}) []interface{} {
+	var values []interface{}
+
+	switch v := fieldValue.(type) {
+	case []string:
+		for _, val := range v {
+			values = append(values, val)
+		}
+	case []interface{}:
+		values = v
```

Comment:
This will not catch any other cases like: `[]int`, `[]float`, etc. This will only catch the case if its type is set as `[]interface{}`. 

Instead of this use the function: `IsArrayOrSlice` that is here: 
https://github.com/siglens/siglens/blob/develop/pkg/utils/sliceutils.go#L70

Reply:
you can directly remove the switch case altogether. And if it is true, then just return the values  and if false, return `nil` like how you do here.

Reply:
Done, I just pushed the change to use the `IsArrayOrSlice` from utils. Also added CI/CD tests and unit tests